src/main/java/hudson/plugins/logparser/LogParserPublisher.java

```diff
@@ -1,232 +1,284 @@
+     *            path to the global parsing rules.
+     * @param useProjectRule
+     *            true if we use a project specific rule.
+     * @param projectRulePath
+     *            path to project specific rules relative to workspace root.
+     */
+    private LogParserPublisher(final boolean unstableOnWarning,
+            final boolean failBuildOnError, final boolean showGraphs,
+            final String parsingRulesPath, final boolean useProjectRule,
+            final String projectRulePath) {
+
+        this.unstableOnWarning = unstableOnWarning;
+        this.failBuildOnError = failBuildOnError;
+        this.showGraphs = showGraphs;
+        this.parsingRulesPath = parsingRulesPath;
+        this.useProjectRule = useProjectRule;
+        this.projectRulePath = projectRulePath;
+    }
+
+    @DataBoundConstructor
+    public LogParserPublisher() {
+        super();
+    }
+
+    @DataBoundSetter
+    public void setUnstableOnWarning(boolean unstableOnWarning) {
+        this.unstableOnWarning = unstableOnWarning;
+    }
+
+    public boolean isUnstableOnWarning() {
+        return unstableOnWarning;
+    }
+
+    @DataBoundSetter
+    public void setFailBuildOnError(boolean failBuildOnError) {
+        this.failBuildOnError = failBuildOnError;
+    }
+
+    public boolean isFailBuildOnError() {
+        return failBuildOnError;
+    }
+
+    @DataBoundSetter
+    public void setShowGraphs(boolean showGraphs) {
+        this.showGraphs = showGraphs;
+    }
+
+    public boolean isShowGraphs() {
+        return showGraphs;
+    }
+
+    @DataBoundSetter
+    public void setUseProjectRule(final JSONObject useProjectRule) {
+        this.useProjectRule = useProjectRule.getBoolean("value");
+        if (useProjectRule.getBoolean("value") && useProjectRule.containsKey("projectRulePath")) {
+            this.projectRulePath = useProjectRule.getString("projectRulePath");
+        } else if (this.useProjectRule && useProjectRule.containsKey("parsingRulesPath")) {
+            this.parsingRulesPath = useProjectRule.getString("parsingRulesPath");
+        }
+    }
+
+    @DataBoundSetter
+    public void setParsingRulesPath(final String parsingRulesPath) {
+        this.parsingRulesPath = parsingRulesPath;
+    }
+
+    @DataBoundSetter
+    public void setProjectRulePath(final String projectRulePath) {
+        this.projectRulePath = projectRulePath;
+    }
+
+    @Override
+    public boolean prebuild(final AbstractBuild<?, ?> build, final BuildListener listener) {
+        return true;
+    }
+
+    @Deprecated
+    public boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher, final BuildListener listener)
+            throws InterruptedException, IOException {
+
+        if (build.getResult() == Result.SUCCESS) {
+            this.perform(build, build.getWorkspace(), launcher, listener);
+        } else {
+            listener.error("Skipping publisher since build result is " + build.getResult());
```

Comment:
Why there is still logic in a deprecated method? Unless I'm missing something, I think it should pass directly to the new implementation.